### DISCUSSION TITLE
Rollback of major upgrades in Bundle does not reinstall MSI

### DISCUSSION THREAD

**Discussion by ManuHArt** [wixtoolset/issues#7076] [Questions] [answered]
Hello everyone,

I have a very basic Wix4 bundle with two MSIs (SetupProject1 and SetupProject2) in a chain. Each package is marked as `vital=yes`.

Each of the MSIs install just two files and are upgraded via `MajorUpgrade` elements which are configured as `Schedule="afterInstallInitialize"`

On upgrade if the second MSI fails, the first MSI is not reinstalled, only the second one, and I just don't know how I have to configure my Bundle / the child MSIs to get the bundle to re-install version 1 of the first MSI.

I've attached archives which contain V1 and V2 of my simple bundle. The V2 will always fail because of a CA in the second MSI.
[MyBootstrapper V1.zip](https://github.com/wixtoolset/issues/files/10195735/MyBootstrapper.V1.zip)
[MyBootstrapper V2 with failing CA.zip](https://github.com/wixtoolset/issues/files/10195781/MyBootstrapper.V2.with.failing.CA.zip)

On upgrade, the following logs are created:
[MyBootstrapper Logs.zip](https://github.com/wixtoolset/issues/files/10195891/MyBootstrapper.Logs.zip)

The thing which really confuses me: in the rollback log of the bundle I see:
`
[122C:0A60][2022-12-09T12:05:14]i201: Planned package: SetupProject1.msi, state: Absent, default requested: Absent, ba requested: Absent, execute: None, rollback: None, default cache strategy: Keep, ba requested strategy: Keep, cache: No, uncache: Yes, dependency: Unregister, expected install registration state: Absent, expected cache registration state: Absent
`
But why "default requested: Absent, ba requested: Absent, execute: None"?
What do I have to change?

I have searched around the web how to fix this but maybe I am just to blind to see the solution right in front of my eyes.

**Comment by robmen** [comment]
There is an issue open about this. You can search for it. 

_Note: It may be closed._

**Comment by rseanhall** [comment]
This is a bug - https://github.com/wixtoolset/issues/issues/7081

**Comment by ManuHArt** [comment, marked as answer]
Finally had the time to re-test.
Fix confirmed in 4.0.0-rc2

Thank you all very much.
